### DISCUSSION TITLE
Docker version

### DISCUSSION THREAD

**Discussion by Meldrak** [benbusby/whoogle-search#174] [Q&A] [answered]
Hi,

I tried docker version of your project and I saw in your screenshots "shopping" in tabs results
I don't have any results or tab for shopping on my side, why?

tor searching option is not working in docker, I have this error message

DEBUG:stem:GETCONF __owningcontrollerprocess (runtime: 0.0003)
INFO:stem:Error while receiving a control message (SocketClosed): received exception "read of closed file"

**Comment by benbusby** [comment, marked as answer]
The shopping tab is disabled until #136 is addressed, since a different set of filtering rules will need to be applied for the shopping tab results. 

I've seen that Tor error message before, but haven't had time to really look into the root cause of it. It seems to not actually have an impact on querying over Tor, since queries can go through fine even when that error pops up. That being said, while Tor searches are technically supported, Google blocks a vast majority of Tor exit nodes and Whoogle will only request a new Tor identity and rerun the query up to 10 times before giving up. If you're seeing an error message on the home page along the lines of "Max attempts exceeded", then the Tor socket is working fine, but none of the queries were accepted by Google.

**Reply by Meldrak** [reply]
Thanks for the explanation, it does not work with my own tor proxy too
